src/einsteinpy/rays/shadow.py

```diff
@@ -0,0 +1,96 @@
+import warnings
+
+import astropy.units as u
+import numpy as np
+from scipy.integrate import quadrature
+from scipy.interpolate import interp1d
+from scipy.optimize import newton
+
+
+class Shadow:
+    """
+    Class for calculating shadow of black holes as seen by distant observer.
+    """
+
+    @u.quantity_input(
+        mass=u.kg, fov=u.km,
+    )
+    def __init__(self, mass, n_rays, fov, limit=0.001):
+        self.mass = mass
+        self.limit = limit
+        self.n_rays = n_rays
+        self.fov = fov
+        self.horizon = 2 * self.mass.value  # To be changed after 0.3.0
+        self.b_crit = 3 * np.sqrt(3) * self.mass
+        self.b = np.linspace(self.b_crit.value, self.fov.value, self.n_rays)
+        self.z = list()
+        self.bfin = list()
+        warnings.filterwarnings("ignore")
```

Comment:
Why this?

Reply:
I am not in position to explain. As I said. I am not satisfied with the performance. So it will be experimental.